neat_core.c

```diff
@@ -486,14 +649,14 @@ static int io_readable(neat_ctx *ctx, neat_flow *flow,
     struct sctp_recvv_rn rn;
     socklen_t infolen = sizeof(struct sctp_recvv_rn);
     int flags = 0;
-#endif
-#endif
+#endif // else !defined(USRSCTP_SUPPORT)
+#endif // defined(SCTP_SUPPORTED)
     neat_log(NEAT_LOG_DEBUG, "%s", __func__);
 
     if (!flow->operations || !flow->operations->on_readable) {
         return READ_WITH_ERROR;
     }
-#if defined(IPPROTO_SCTP)
+#if defined(SCTP_SUPPORTED)
     if ((flow->sockProtocol == IPPROTO_SCTP) &&
```

Comment:
Ok, that does make sense for this particular code path. In the bigger picture, I'm a bit worried that things will still be broken elsewhere because of assumptions around what to do when SCTP support is detected or not.

Is there any consensus on how this should work? I see for example that on OSX, HE thinks it can try SCTP, although that isn't really supported. That's different from NetBSD, where it only detects TCP.

Also, what happens if several different SCTP implementations are available?
Right now, all code paths will always prefer USRSCTP if that is compiled in if I understand the code correctly.